### PR TITLE
home screen UI and database rebuild (WIP)

### PR DESCRIPTION
**Pull request by PaloneAppEnjoyer**
image
image

### CODE REVIEW

File: app/src/main/java/com/palone/planahead/screens/home/HomeScreenViewModel.kt [hunk 1]
```diff
@@ -1,52 +1,101 @@
...
+    fun updateMockTaskPriority(taskPriority: TaskPriority) {
+        _uiState.update { _uiState.value.copy(mockAlertTaskPriority = taskPriority) }
     }
 
-    fun deleteTask(task: Task) {
+    fun createDatabaseEntry(
+        description: String,
+        alertTypes: List<AlertType>,
+        alertTriggers: List<AlertTrigger>,
+        alertTriggerData: String
```
[the remark this answers is not in the document]
Comment (PaloneAppEnjoyer) [reply app/src/main/java/com/palone/planahead/screens/home/HomeScreenViewModel.kt:57]: so how else can i get JSON data if it's based on AlertTrigger?

File: app/src/main/java/com/palone/planahead/screens/home/HomeScreenViewModel.kt [hunk 2]
```diff
@@ -1,52 +1,101 @@
...
-            taskRepository.delete(task)
+            val today = LocalDateTime.now().format(DateTimeFormatter.ISO_LOCAL_DATE_TIME)
+            taskAndAlertBuilder.updateTask(
+                Task(
+                    description = description,
+                    addedDate = today,
+                    isCompleted = false
+                )
+            )
+            alertTypes.forEach { alertType -> //TODO figure out better way to add lists to database
```
[the remark this answers is not in the document]
Comment (PaloneAppEnjoyer) [reply app/src/main/java/com/palone/planahead/screens/home/HomeScreenViewModel.kt:69]: We are saving all AlertTypes and Triggers what are selected by the user, this function is called with uiState mockTask values

File: app/src/main/java/com/palone/planahead/data/TaskAndAlertBuilder.kt [hunk 3]
```diff
@@ -0,0 +1,43 @@
+package com.palone.planahead.data
+
+import com.palone.planahead.data.database.TaskRepository
+import com.palone.planahead.data.database.alert.Alert
+import com.palone.planahead.data.database.alert.properties.AlertTrigger
+import com.palone.planahead.data.database.alert.properties.AlertType
+import com.palone.planahead.data.database.task.Task
+
+class TaskAndAlertBuilder(private val repository: TaskRepository) {
```
[the remark this answers is not in the document]
Comment (PaloneAppEnjoyer) [reply app/src/main/java/com/palone/planahead/data/TaskAndAlertBuilder.kt:9]: This class helps in composing and sending to database tasks and its alerts, its not just a initial database creator

[the remark this answers is not in the document]
Comment (Albitos) [reply app/src/main/java/com/palone/planahead/data/TaskAndAlertBuilder.kt:9]: Can't we just create task and alerts directly? I don't see any real logic there.

File: app/src/main/java/com/palone/planahead/screens/home/HomeScreenViewModel.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (Albitos) [reply app/src/main/java/com/palone/planahead/screens/home/HomeScreenViewModel.kt:57]: It should be of type `AlertTriggerData`. We should have several types inheriting from it - e.g. `LocationTriggerData`, `TimeTriggerData`, etc.

**Pull request merged**